cmd/shellExecute.go

```diff
@@ -79,3 +95,25 @@ func runShellExecute(config *shellExecuteOptions, telemetryData *telemetry.Custo
 
 	return nil
 }
+
+func downloadScript(config *shellExecuteOptions, utils shellExecuteUtils, url string) (string, error) {
+	header := http.Header{}
+	if len(config.GithubToken) > 0 {
+		header = http.Header{"Authorization": []string{"Token " + config.GithubToken}}
+		header.Set("Accept", "application/vnd.github.v3.raw")
+	}
+
+	log.Entry().Infof("downloading script : %v", url)
+	fileNameParts := strings.Split(url, "/")
+	fileName := fileNameParts[len(fileNameParts)-1]
+	err := utils.DownloadFile(url, filepath.Join(".pipeline", fileName), header, []*http.Cookie{})
```

Comment:
why don't you download to the working dir?

We cannot assume that `.pipeline` folder exists.

Reply:
`DownloadFile` make sure the parent directory is created if its non existent . downloading it in the current directory would require to send back the string as `./fileName.sh` to make it runnable in the current directory and wanted to keep the solution clean without `"."` :)